guacamole-common-js/src/main/webapp/modules/KeyEventInterpreter.js

```diff
@@ -424,6 +452,29 @@ Guacamole.KeyEventInterpreter = function KeyEventInterpreter(batchSeperation, st
                 var keyText;
                 var typed;
 
+                // If this key should be both printed and rendered
+                if (keyDefinition.value != null && keyDefinition.printAndRender) {
+
+                    // The printed name of the key
+                    var printedValue = getPrintedKeyName(keyDefinition);
+
+                    // The directly rendered value
+                    var renderedValue = keyDefinition.value;
+
+                    // Display the name first, followed by the value
+                    currentBatch.simpleValue += printedValue;
+                    currentBatch.simpleValue += renderedValue;
+
+                    // Add pair of key events in the same order, with the name
+                    // displayed as non-typed to ensure proper styling
+                    currentBatch.events.push(new Guacamole.KeyEventInterpreter.KeyEvent(
+                        printedValue, false, relativeTimestap));
+                    currentBatch.events.push(new Guacamole.KeyEventInterpreter.KeyEvent(
+                        renderedValue, true, relativeTimestap));
```

Comment:
I don't think individual events should possibly be split into multiple events purely for the sake of rendering, as it's not guacamole-common-js that should dictate that rendering.

I think we should instead settle on a structure that represents both the expected typed result (if any) of a logical event and the keys pressed to produce that result, leaving it up to the implementation to decide how that is ultimately rendered.

Reply:
So currently the `KeyEventIntepreter` is documented to "produce human readable text batches". 

It sounds like you're advocating removing all of the rendering logic out of this service (i.e. everything that produces the `text` or `simpleValue` fields, modifier/shortcut tracking, probably even batch separation). This would be probably all move to a service in the webapp. 

I think that separation of concerns makes a lot of sense, though I was unsure how much heavy handed I should be in changing the way the interpreter works.

Reply:
It doesn't look like we've released this code yet, so it seems like it should be ok.

Reply:
@mike-jumper How does this look?